mindsdb/integrations/handlers/eventstoredb_handler/eventstoredb_handler.py

```diff
@@ -0,0 +1,212 @@
+from mindsdb_sql.parser.ast.base import ASTNode
+from mindsdb_sql.parser.ast import Select
+from mindsdb.utilities import log
+from mindsdb.integrations.libs.base import DatabaseHandler
+from mindsdb.integrations.libs.response import (
+    HandlerStatusResponse as StatusResponse,
+    HandlerResponse as Response,
+    RESPONSE_TYPE
+)
+
+from .utils.helpers import *
+import requests
+import pandas as pd
+from mindsdb_sql import parse_sql, get_lexer_parser
+
+
+class EventStoreDB(DatabaseHandler):
+    """
+    Handler for EventStoreDB
+    The handler uses the Atom Pub Over HTTP of EventStoreDB.
+    This means that you need to enable AtomPuvOverHTTP on your EventStoreDB instance if you are using v20+
+
+    Why not gRPC? At the moment we cannot use https://pypi.org/project/esdbclient/ which uses the gRPC endpoint
+    because mysql-connector-python 8.0.32 requires protobuf<=3.20.3,>=3.11.0, but esdbclient
+    requires protobuf 4.22.1.
+
+    Why not TCP? At the moment https://github.com/epoplavskis/photon-pump only works in insecure mode and
+    configuration is limited.
+
+    Third reason, there is no official Python client at the moment of writing of this handler.
+    But once there is better ESDB Python support for gRPC, we should move this integration from AtomPub to gRPC.
+    """
+
+    name = 'eventstoredb'
+    # defaults to an insecure localhost single node
+    scheme = 'http'
+    host = 'localhost'
+    port = '2113'
+    is_connected = None
+    basic_url = ""
+    read_batch_size = 500  # should be adjusted based on use case
+    headers = {
+        'Accept': 'application/json',
+        'ES-ResolveLinkTo': "True"
+    }
+    tlsverify = False
+
+    def __init__(self, name, **kwargs):
+        super().__init__(name)
+        self.parser = parse_sql
+        connection_data = kwargs['connection_data']
+        self.host = connection_data.get('host')
+        if connection_data.get('tls') is not None and isinstance(connection_data.get('tls'), bool) \
+                and connection_data.get('tls'):
+            self.scheme = 'https'
+        if connection_data.get('port') is not None and isinstance(connection_data.get('port'), int):
+            self.port = connection_data.get('port')
+        if connection_data.get('page_size') is not None:
+            if isinstance(connection_data.get('page_size'), int) and connection_data.get('page_size') > 0:
+                self.read_batch_size = connection_data.get('page_size')
+        log.logger.debug(f'scheme: {self.scheme} host: {self.host}  port:{self.port} page: {self.read_batch_size}')
+        if connection_data.get('username') is not None and connection_data.get('password') is not None:
+            if isinstance(connection_data.get('username'), str) and isinstance(connection_data.get('password'), str):

```

Comment:
This are long lines that will fail when we lint the code. Please make them shorter

Reply:
shortened them